u-mode-api/src/lib.rs

```diff
@@ -144,73 +144,65 @@ impl TryFrom<u64> for UmodeOp {
 pub struct UmodeRequest {
     /// The operation requested.
     pub op: UmodeOp,
-    /// Optional start of mapped area accessible as read-only.
-    pub in_addr: Option<u64>,
-    /// If in_addr is valid, length of the area acessible as read-only.
-    pub in_len: usize,
-    /// Optional start of mapped area accessible as read-write.
-    pub out_addr: Option<u64>,
-    /// If in_addr is valid, length of the area acessible as read-write.
-    pub out_len: usize,
+    /// Arguments of the operation.
+    pub args: [u64; 7],
```

Comment:
Side note suggestion for consideration, not a request for code changes: With my rust data modeling nerd hat on, I would rather make these data fields in the UmodeOp enum constants like so:

```
pub enum UmodeOp {
    // ...
    MemCopy(u64 dest_addr, u64 src_addr, u64 len),
}
```

And then you can have the appropriate TryIntoRegisters implementations to encode/decode them.

Reply:
I thought about this. What you suggest is indeed the rust-est way to do it. I just felt that this was more concise and explained easier to people reading it how the API maps to it. No strong opinions on it.

Reply:
Right, so I implemented this -- I already did something similar for hypcalls right below this -- and yes, it is much better.

Updated the PR. Thank you.

Reply:
Nice, I like it!